frontend/src/pages/articles/articles-feed.tsx

```diff
@@ -1,12 +1,36 @@
+import {
+  useAppDispatch,
+  useAppSelector,
+  useEffect,
+} from '~/libs/hooks/hooks.js';
+import { type UserDetailsResponseDto } from '~/packages/users/users.js';
+import { actions as articlesActions } from '~/slices/articles/articles.js';
+
 import { ArticleCard } from './components/components.js';
-import { MOCKED_ARTICLES, MOCKED_USER } from './libs/constants.js';
-
-const ArticlesFeed: React.FC = () => (
-  <>
-    {MOCKED_ARTICLES.map((article) => (
-      <ArticleCard key={article.id} article={article} user={MOCKED_USER} />
-    ))}
-  </>
-);
+import { MOCKED_REACTIONS, MOCKED_TAGS } from './libs/constants.js';
+
+const ArticlesFeed: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { articles } = useAppSelector(({ articles }) => articles);
+
+  useEffect(() => {
+    void dispatch(articlesActions.fetchAll());
+  }, [dispatch]);
+
+  return (
+    <>
+      {articles?.every((article) => article.author) &&
+        articles.map((article) => (
```

Comment:
I guess we can place the logic for checking article author existence inside a single map loop, if there’s no author we return null, otherwise ArticleCard
```js
{
  articles.map(article => article.author ? <ArticleCard {…your props} /> : null)
}

```